provider/github/watcher.go

```diff
@@ -177,9 +180,14 @@ func (w *Watcher) processRepoPRs(
 ) (time.Duration, error) {
 	resp, prs, err := w.doPRListRequest(ctx, c, repo.Owner, repo.Name)
 	if ErrGitHubAPI.Is(err) {
-		ctxlog.Get(ctx).With(log.Fields{
+		if w.lastErrPR[repo] == nil {
+			w.lastErrPR[repo] = &errThrottlerState{}
+		}
+
+		newErrThrottlerLogger(ctxlog.Get(ctx), w.lastErrPR[repo]).With(log.Fields{
 			"repository": repo.FullName, "response": resp,
-		}).Errorf(err, "request for PR list failed")
+		}).Errorf(fmt.Errorf(err.Error()), "request for PR list failed")
+
```

Comment:
```go
fmt.Errorf(err.Error())
```
I'm confused. Why do you need it?

Reply:
To remove the stack trace.

With this change we log:
```
[2018-12-26T09:43:08.663549151Z] ERROR request for PR list failed app=lookoutd error=github api error: GET https://api.github.com/repos/nope/nope/pulls: 404 Not Found [] repository=nope/nope response=github.Rate{Limit:5000, Remaining:4998, Reset:github.Timestamp{2018-12-26 10:43:08 +0000 WET}}
```

With `.Errorf(err, `, we log:
```
[2018-12-26T09:44:38.594569159Z] ERROR request for PR list failed app=lookoutd error=github api error: GET https://api.github.com/repos/nope/nope/pulls: 404 Not Found []

github.com/src-d/lookout/provider/github.(*Watcher).doPRListRequest
	/home/cmartin/go/src/github.com/src-d/lookout/provider/github/watcher.go:308
github.com/src-d/lookout/provider/github.(*Watcher).processRepoPRs
	/home/cmartin/go/src/github.com/src-d/lookout/provider/github/watcher.go:181
github.com/src-d/lookout/provider/github.(*Watcher).processRepoPRs-fm
	/home/cmartin/go/src/github.com/src-d/lookout/provider/github/watcher.go:132
github.com/src-d/lookout/provider/github.(*Watcher).watchLoop
	/home/cmartin/go/src/github.com/src-d/lookout/provider/github/watcher.go:151
runtime.goexit
	/usr/lib/go-1.11/src/runtime/asm_amd64.s:1333 repository=nope/nope response=github.Rate{Limit:5000, Remaining:4993, Reset:github.Timestamp{2018-12-26 10:43:08 +0000 WET}}
```

It's an API error in a known place ("request for PR list failed"), so there is not need to flood the logs with extra debug information.

Reply:
Maybe use Warn level then? If this error is not "real" error...

Reply:
Why is it not a "real" error?

Reply:
Errors should be rendered with stack trace according to our logging library.
You work around it using
```go
fmt.Errorf(err.Error())
```
which is a hack that makes the error "not-really-error" but a more a warning in terms of our logging lib

Reply:
Well, the way I see it, it's not a panic, but it's still a "real" error. Missconfiguration error, connectivity error, or auth error... but something the user will be interested in knowing and fixing. A warning level might be overlooked too easily.

I don't think removing the stack trace makes it a hack. If we didn't log it here and instead returned, it could be `fmt.Errorf("request for events list failed: %s", err)`. Just like you can add relevant information to an error, I don't see a problem removing extra information that we know is not relevant.

Reply:
```go
fmt.Errorf(err.Error())
```
does look like a hack and WTF. I'm pretty sure anybody reading this code will have the same question as I did.

I don't mind to merge it now. But it does NOT look right.

Reply:
a cleaner version of the same code (though I have doubts it's should be written like that at all):
```go
UserError(err)
```

Reply:
also better than right now 👍

Reply:
done in 994c357